lualib/lua_scanners/avast_rest.lua

```diff
@@ -0,0 +1,354 @@
+    log_clean = false,
+    tmpdir = '/tmp',
+    retransmits = 1,
+    cache_expire = 7200, -- expire redis in 2h
+    message = '${SCANNER}: virus message found: "${VIRUS}"',
+    detection_category = "virus",
+    default_score = 1,
+    action = false,
+    scan_mime_parts = true,
+    scan_text_mime = false,
+    scan_image_mime = false,
+    warnings_as_threat = false,
+    -- https://repo.avcdn.net/linux-av/doc/avast-techdoc.pdf
+    parameter = {
+      --email = false,
+      --full = false,
+      archives = true,
+      --pup = false,
+      --heuristics = 40,
+      --detections = false,
+    },
+  }
+
+  default_conf = lua_util.override_defaults(default_conf, opts)
+
+  if not default_conf.prefix then
+    default_conf.prefix = 'rs_' .. default_conf.name .. '_'
+  end
+
+  if not default_conf.log_prefix then
+    if default_conf.name:lower() == default_conf.type:lower() then
+      default_conf.log_prefix = default_conf.name
+    else
+      default_conf.log_prefix = default_conf.name .. ' (' .. default_conf.type .. ')'
+    end
+  end
+
+  if not default_conf.servers and default_conf.socket then
+    default_conf.servers = default_conf.socket
+  end
+
+  if not default_conf.servers then
+    rspamd_logger.errx(rspamd_config, 'no servers defined')
+
+    return nil
+  end
+
+  default_conf.upstreams = upstream_list.create(rspamd_config,
+      default_conf.servers,
+      default_conf.default_port)
+
+  if default_conf.upstreams then
+    lua_util.add_debug_alias('antivirus', default_conf.name)
+    return default_conf
+  end
+
+  rspamd_logger.errx(rspamd_config, 'cannot parse servers %s',
+      default_conf['servers'])
+  return nil
+end
+
+local function avast_rest_check(task, content, digest, rule, maybe_part)
+  local function avast_rest_check_uncached()
+
+    local function make_url(addr, filename)
+      local url
+      local suffix = '/v1/scan?'
+
+      local param_tbl = {}
+      if rule.use_files then
+        rule.parameter['path'] = filename
+      else
+        rule.parameter['filename'] = filename
+      end
+      if not rule.parameter.email and not rule.scan_mime_parts then
+        rule.parameter['email'] = true
+      end
+      for k,v  in pairs(rule.parameter) do
+        table.insert(param_tbl, string.format('%s=%s', k, tostring(v)))
+      end
+
+      if rule.use_https then
+        url = string.format('https://%s:%d%s%s', tostring(addr), addr:get_port(), suffix,
+          table.concat(param_tbl, '&'))
+      else
+        url = string.format('http://%s:%d%s%s', tostring(addr), addr:get_port(), suffix,
+          table.concat(param_tbl, '&'))
+      end
+      return url
+    end
+
+    local upstream = rule.upstreams:get_upstream_round_robin()
+    local addr = upstream:get_addr()
+    local retransmits = rule.retransmits
+
+    local request_data = {
+      task = task,
+      timeout = rule.timeout,
+    }
+
+    local filename
+    if rule.use_files then
+      filename =  string.format('%s/%s.tmp',
```

Comment:
```suggestion
      filename =  string.format('%s/%s.tmp.ravast',
```

To help detecting file leaks if any...